models/readme.md

```diff
@@ -1 +1,164 @@
+
+```
+cd MLAGILITY_ROOT/models # MLAGILITY_ROOT is where you cloned mlagility
+benchit benchmark linear.py --search-dir selftest
+```
+
+You can also evaluate an entire corpus with a command like this:
+```
+cd MLAGILITY_ROOT/models # MLAGILITY_ROOT is where you cloned mlagility
+benchit benchmark --all --search-dir selftest
+```
+
+You can aggregate all of the benchmarking results from your `mlagility cache` into a CSV file with:
+
+```
+benchit report
+```
+
+If you want to only report on a subset of models, we recommend benchmarking those models into a specific cache directory:
+
+```
+# Benchmark the models into a specific cache directory
+benchit benchmark --all --search-dir selftest --cache-dir selftest_results
+
+# Report the results from the `selftest_results` cache
+benchit report --cache-dir selftest_results
+```
+
+## Model Template
+
+Each model in the MLAgility benchmark follows a template to keep things consistent. This template is meant to be as simple as possible. The models themselves should also be instantiated and called in a completely vendor-neutral manner.
+
+### The model.py File
+
+Each model in the MLAgility benchmark is hosted in a `model.py` file. This file is a Python script that must meet the following requirements:
+1. Instantiate at least one model and invoke it against some inputs. The size and shape of the inputs will be used for benchmarking.
+1. Provide a docstring that provides information about where the model was sourced from.
+
+The file should be named according to the model within. For example, our file for the BERT model is named `bert.py`.
+
+Each `model.py` can optionally include a set of [labels](#labels) and [parameters](#parameters).
+
+### Labels
+
+MLAgility uses labels to help organize the results data. Labels must be in the first line of the Python file and start with `# labels: `
+
+Each label must have the format `key::value1,value2,...`
+
+Example:
+
+```
+# labels: author::google test_group::daily,monthly
+```
+     
+Labels are saved in your cache directory and can later be retrieved using the function `mlagility.common.labels.load_from_cache()`, which receives the `cache_dir` and `build_name` as inputs and returns the labels as a dictionary. 
+
+### Parameters
+
+MLAgility supports parameterizing models so that you can sweep over interesting properties such as batch size, sequence length, image size, etc. The `mlagility.parser.parse()` method parses a set of standardized parameters that we have defined and provides them to the model as it is instantiated.
+
+For example, this code would retrieve user-defined `batch_size` and `max_seq_length` (maximum sequence length) parameters.
+
+```
+from mlagility.parser import parse
+parse(["batch_size", "max_seq_length"])
+```
+
+You can pass parameters into a benchmarking run with the `--script-args` argument to `benchit`. For example, the command:
```

Comment:
Might be useful to explain that those CLI arguments may also be used outside of benchit

Example `bert.py --batch-size 8`

Reply:
Added!